modules/apps/frontend-editor/frontend-editor-alloyeditor-web/src/main/java/com/liferay/frontend/editor/alloyeditor/web/internal/AlloyEditorEditor.java

```diff
@@ -28,47 +25,12 @@
  * @author Raymond Augé
  * @author Roberto Díaz
  */
-@Component(
-	property = "name=alloyeditor",
-	service = {Editor.class, EditorRenderer.class}
-)
-public class AlloyEditorEditor implements Editor, EditorRenderer {
-
-	@Override
-	public String getAttributeNamespace() {
-		return AlloyEditorConstants.ATTRIBUTE_NAMESPACE;
-	}
-
-	@Override
-	public String[] getJavaScriptModules() {
-		return new String[] {"liferay-alloy-editor"};
-	}
-
-	@Override
-	public String getJspPath() {
-		return "/alloyeditor.jsp";
-	}
-
-	@Override
-	public String getName() {
-		return _name;
-	}
-
-	@Override
-	public String getResourcesJspPath() {
-		return "/resources.jsp";
-	}
-
-	@Override
-	public String getResourceType() {
-		return PortalWebResourceConstants.RESOURCE_TYPE_EDITOR_ALLOYEDITOR;
-	}
+@Component(property = "name=alloyeditor", service = Editor.class)
+public class AlloyEditorEditor extends BaseAlloyEditor {
 
 	@Activate
 	protected void activate(Map<String, Object> properties) {
-		_name = (String)properties.get("name");
+		setName((String)properties.get("name"));
```

Comment:
Hey @MrJohn1911, please create a constant called `EDITOR_NAME` with value `alloyeditor` and implement the `getName` and set the Component property as well. Also, get rid of the `setName` method. 

Please apply for the other cases as well. Thx